src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -1205,6 +1210,9 @@ private Task DestroyActivations(List<ActivationData> list)
                         directory.InvalidateCacheEntry(activationData.Address);
 
                         RerouteAllQueuedMessages(activationData, null, "Finished Destroy Activation");
+
+                        activationData.ServiceScope?.Dispose();
+                        activationData.ServiceScope = null;
```

Comment:
I don't know for sure, but since we are creating the ServiceScope externally and then assigning it, then I think it's fine to also Dispose externally. Otherwise I think I'd have to change the logic so that the ActivationData itself creates the scope. Thoughts? is it worse it?